File: frontend/src/types/schedule.ts
/**
 * スケジュール型定義
 */

export interface Schedule {
  scheduleId: string;
  userId: string;
  date: string; // YYYY-MM-DD形式
  isDone: boolean;
  recipeId: string;
  recipeTitle: string;
  memo?: string;
  createdAt: string;
}

export interface CreateScheduleRequest {
  date: string; // YYYY-MM-DD形式
  recipeId: string;
  recipeTitle: string;
  memo?: string;
}

export interface UpdateScheduleRequest {
  memo?: string;
}

export interface ScheduleSearchParams {
  startDate: string; // YYYY-MM-DD形式
  endDate: string; // YYYY-MM-DD形式
}

export interface ScheduleState {
  schedules: Schedule[];
  loading: boolean;
  error: string | null;
}

File: backend-python/app/models/schedule.py
from dataclasses import dataclass, field
from datetime import date, datetime


@dataclass
class Schedule:
    schedule_id: str
    user_id: str
    date: date
    recipe_id: str
    recipe_title: str
    is_done: bool = False
    memo: str = ""
    created_at: datetime = field(default_factory=datetime.now)
